File: swift/secret-handshake/Sources/SecretHandshake/SecretHandshake.swift
import Foundation

struct SecretHandshake {
    var commands = [String]()

    init(_ input: Int) {
        for (index, value) in String(input, radix: 2).reversed().enumerated() {
            if value == "1" {
                switch index {
                case 0:
                    commands += ["wink"]
                case 1:
                    commands += ["double blink"]
                case 2:
                    commands += ["close your eyes"]
                case 3:
                    commands += ["jump"]
                case 4:
                    commands = commands.reversed()
                default:
                    break
                }
            }
        }
    }
}

File: go/secret-handshake/secret_handshake.go
package secret

// Handshake returns a sequence of strings in the secret code.
func Handshake(n uint) []string {
	var h []string
	if n >= 32 || n == 0 {
		return h
	}

	if n&1 == 1 {
		h = append(h, "wink")
	}
	if n&2 == 2 {
		h = append(h, "double blink")
	}
	if n&4 == 4 {
		h = append(h, "close your eyes")
	}
	if n&8 == 8 {
		h = append(h, "jump")
	}
	if n&16 == 16 {
		for left, right := 0, len(h)-1; left < right; left, right = left+1, right-1 {
			h[left], h[right] = h[right], h[left]
		}
	}

	return h
}
